packages/apm/src/integrations/tracing.ts

```diff
@@ -209,8 +219,17 @@ export class Tracing implements Integration {
           event.timestamp - event.start_timestamp < 0);
 
       if (Tracing.options.maxTransactionDuration !== 0 && event.type === 'transaction' && isOutdatedTransaction) {
-        logger.log('[Tracing] Discarded transaction since it maxed out maxTransactionDuration');
-        return null;
+        Tracing._log('[Tracing] Discarded transaction since it maxed out maxTransactionDuration');
+        if (event.contexts && event.contexts.trace) {
+          event.contexts.trace = {
+            ...event.contexts.trace,
+            status: SpanStatus.DeadlineExceeded,
+          };
+          event.tags = {
+            ...event.tags,
+            maxTransactionDurationExceeded: 'true',
```

Comment:
`true` will be stringified anyway, so we can use `boolean` here directly

Reply:
Tags only allow string values :|